Build failure: xone

### Steps To Reproduce

Steps to reproduce the behavior:
1. build xone

### Build log

```

nix log /nix/store/cbcv9vfynssj9bl0mrxr5zggj9c8rr8m-xone-0.3.drv
warning: The interpretation of store paths arguments ending in `.drv` recently changed. If this command is now failing try again with '/nix/store/cbcv9vfynssj9bl0mrxr5zggj9c8rr8m-xone-0.3.drv^*'
@nix { "action": "setPhase", "phase": "unpackPhase" }
Running phase: unpackPhase
unpacking source archive /nix/store/sxvbr4xpsz2w9zd2mxqsdxl3zkcz6wfi-source
source root is /build/source
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: patchPhase
applying patch /nix/store/15qds5bbdym41di9g70krvi9d4wjbqyx-kernel-6.3.patch
patching file bus/bus.c
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: configurePhase
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
Running phase: buildPhase
build flags: SHELL=/nix/store/4vzal97iq3dmrgycj8r0gflrh51p8w1s-bash-5.2p26/bin/bash -C /nix/store/q3k5l3bf08s4wa5im3lj63vivqj4ml4y-linux-6.8-dev/lib/modules/6.8.0-zen1/build M=\$\(sourceRoot\) VERSION=0.3>
make: Entering directory '/nix/store/q3k5l3bf08s4wa5im3lj63vivqj4ml4y-linux-6.8-dev/lib/modules/6.8.0-zen1/build'
  CC [M]  /build/source/transport/wired.o
  LD [M]  /build/source/xone-wired.o
  CC [M]  /build/source/transport/dongle.o
/build/source/transport/dongle.c:970:10: error: 'struct usb_driver' has no member named 'drvwrap'
  970 |         .drvwrap.driver.shutdown = xone_dongle_shutdown,
      |          ^~~~~~~
/build/source/transport/dongle.c:970:36: error: initialization of 'const struct attribute_group **' from incompatible pointer type 'void (*)(struct device *)' [-Werror=incompatible-pointer-types]
  970 |         .drvwrap.driver.shutdown = xone_dongle_shutdown,
      |                                    ^~~~~~~~~~~~~~~~~~~~
/build/source/transport/dongle.c:970:36: note: (near initialization for 'xone_dongle_driver.dev_groups')
cc1: some warnings being treated as errors
make[2]: *** [/nix/store/q3k5l3bf08s4wa5im3lj63vivqj4ml4y-linux-6.8-dev/lib/modules/6.8.0-zen1/source/scripts/Makefile.build:243: /build/source/transport/dongle.o] Error 1
make[1]: *** [/nix/store/q3k5l3bf08s4wa5im3lj63vivqj4ml4y-linux-6.8-dev/lib/modules/6.8.0-zen1/source/Makefile:1919: /build/source] Error 2
make: *** [/nix/store/q3k5l3bf08s4wa5im3lj63vivqj4ml4y-linux-6.8-dev/lib/modules/6.8.0-zen1/source/Makefile:240: __sub-make] Error 2
make: Leaving directory '/nix/store/q3k5l3bf08s4wa5im3lj63vivqj4ml4y-linux-6.8-dev/lib/modules/6.8.0-zen1/build'
lines 1-32/32 (END)

```

### Additional context



### Notify maintainers

@rhysmdnz 

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.7.9-zen1, NixOS, 24.05 (Uakari), 24.05.20240309.3030f18`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/nra828scc8qs92b9pxra5csqzffb6hpl-source`
---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

I am not able to build either, but my error looks diffrent:
```
building the system configuration...
error: builder for '/nix/store/qwxaapzybz6zc2sac33p87rkhdl1anws-xone-0.3.drv' failed with exit code 2;
       last 10 log lines:
       >       |          ^~~~~~~
       > /build/source/transport/dongle.c:970:36: error: initialization of 'const struct attribute_group **' from incompatible pointer type 'void (*)(struct device *)' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Werror=incompatible-pointer-types8;;]
       >   970 |         .drvwrap.driver.shutdown = xone_dongle_shutdown,
       >       |                                    ^~~~~~~~~~~~~~~~~~~~
       > /build/source/transport/dongle.c:970:36: note: (near initialization for 'xone_dongle_driver.dev_groups')
       > cc1: some warnings being treated as errors
       > make[2]: *** [/nix/store/46l0ym1wzfffmslhni6awbkd3wy13vrp-linux-6.8-dev/lib/modules/6.8.0/source/scripts/Makefile.build:243: /build/source/transport/dongle.o] Error 1
       > make[1]: *** [/nix/store/46l0ym1wzfffmslhni6awbkd3wy13vrp-linux-6.8-dev/lib/modules/6.8.0/source/Makefile:1921: /build/source] Error 2
       > make: *** [/nix/store/46l0ym1wzfffmslhni6awbkd3wy13vrp-linux-6.8-dev/lib/modules/6.8.0/source/Makefile:240: __sub-make] Error 2
       > make: Leaving directory '/nix/store/46l0ym1wzfffmslhni6awbkd3wy13vrp-linux-6.8-dev/lib/modules/6.8.0/build'
       For full logs, run 'nix log /nix/store/qwxaapzybz6zc2sac33p87rkhdl1anws-xone-0.3.drv'.
error: 1 dependencies of derivation '/nix/store/g60dgqvqygwq194nz7c4fkyfwfs3mqm9-kernel-modules.drv' failed to build
error (ignored): error: cannot unlink '/tmp/nix-build-nvidia-x11-550.54.14-6.8.drv-0/NVIDIA-Linux-x86_64-550.54.14/kernel': Directory not empty
error: 1 dependencies of derivation '/nix/store/hk18ylzhlpbcpbrgkgnw4kahwcj201ak-linux-6.8-modules.drv' failed to build
error: 1 dependencies of derivation '/nix/store/1f5lchdxfvjlcyg3l0mvn9qbxfz11i5z-nixos-system-Ni-24.05.20240312.0ad13a6.drv' failed to build

``` 
```bash
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.7.9, NixOS, 24.05 (Uakari), 24.05.20240309.3030f18`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixos-23.05"`
 - nixpkgs: `/nix/store/nra828scc8qs92b9pxra5csqzffb6hpl-source`
```

--- a/pkgs/os-specific/linux/xone/default.nix
+++ b/pkgs/os-specific/linux/xone/default.nix
@@ -1,27 +1,18 @@
 { stdenv, lib, fetchFromGitHub, kernel, fetchurl, fetchpatch }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "xone";
-  version = "0.3";
+  version = "0.3-unstable-2024-03-16";
 
   src = fetchFromGitHub {
     owner = "medusalix";
-    repo = pname;
-    rev = "refs/tags/v${version}";
-    sha256 = "sha256-h+j4xCV9R6hp9trsv1NByh9m0UBafOz42ZuYUjclILE=";
+    repo = "xone";
+    rev = "948d2302acdd6333295eaba4da06d96677290ad3";
+    hash = "sha256-srAEw1ai5KT0rmVUL3Dut9R2mNb00AAZVCcINikh2sM=";
   };
 
-  patches = [
-    # Fix build on kernel 6.3
-    (fetchpatch {
-      name = "kernel-6.3.patch";
-      url = "https://github.com/medusalix/xone/commit/bbf0dcc484c3f5611f4e375da43e0e0ef08f3d18.patch";
-      hash = "sha256-A2OzRRk4XT++rS6k6EIyiPy/LJptvVRUxoP7CIGrPWU=";
-    })
-  ];
-
   setSourceRoot = ''
-    export sourceRoot=$(pwd)/${src.name}
+    export sourceRoot=$(pwd)/${finalAttrs.src.name}
   '';
 
   nativeBuildInputs = kernel.moduleBuildDependencies;
@@ -30,7 +21,7 @@ stdenv.mkDerivation rec {
     "-C"
     "${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
     "M=$(sourceRoot)"
-    "VERSION=${version}"
+    "VERSION=${finalAttrs.version}"
   ];
 
   buildFlags = [ "modules" ];
@@ -45,4 +36,4 @@ stdenv.mkDerivation rec {
     platforms = platforms.linux;
   };
 }
-
+)

